pkg/ociinstaller/ocidownloader.go

```diff
@@ -27,27 +35,87 @@ func NewOciDownloader() *ociDownloader {
 	}
 }
 
-/* 
+/*
 Pull downloads the image from the given `ref` to the supplied `destDir`
 
 Returns
 
 	imageDescription, configDescription, config, imageLayers, error
 */
 func (o *ociDownloader) Pull(ctx context.Context, ref string, mediaTypes []string, destDir string) (*ocispec.Descriptor, *ocispec.Descriptor, []byte, []ocispec.Descriptor, error) {
-	log.Println("[TRACE] ociDownloader.Pull:", "pulling", ref)
-	fileStore := content.NewFileStore(destDir)
+	split := strings.Split(ref, ":")
+	tag := split[len(split)-1]
+	log.Println("[TRACE] ociDownloader.Pull:", "preparing to pull ref", ref, "tag", tag, "destDir", destDir)
+
+	// Create the target file store
+	memoryStore := memory.New()
+	fileStore, err := file.NewWithFallbackStorage(destDir, memoryStore)
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
 	defer fileStore.Close()
 
-	hybridStore := newHybridStore(fileStore)
-	pullOpts := []oras.PullOpt{
-		oras.WithAllowedMediaTypes(append(mediaTypes, MediaTypeConfig, MediaTypePluginConfig)),
-		oras.WithPullEmptyNameAllowed(),
+	// Connect to the remote repository
+	repo, err := remote.NewRepository(ref)
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
+
+	// Get credentials from the docker credentials store
+	storeOpts := credentials.StoreOptions{}
+	credStore, err := credentials.NewStoreFromDocker(storeOpts)
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
+
+	// Prepare the auth client for the registry and credential store
+	repo.Client = &auth.Client{
+		Client:     retry.DefaultClient,
+		Cache:      auth.DefaultCache,
+		Credential: credentials.Credential(credStore), // Use the credential store
 	}
-	desc, layers, err := oras.Pull(ctx, o.resolver, ref, hybridStore, pullOpts...)
+
+	// Copy from the remote repository to the file store
+	log.Println("[TRACE] ociDownloader.Pull:", "pulling...")
+
+	copyOpt := oras.DefaultCopyOptions
+	// TODO: use WithTargetPlatform to limit downloads
```

Comment:
Not sure if this is desired or not as there are checks later to ensure binaries for several platforms exist.

Reply:
If this limits the layers downloaded, then we should probably include it

Reply:
It apparently require more changes than I expected.

As a test, just adding this:

```
```

Results in:

```
steampipe plugin install csv:latest
2023-06-26 15:09:57.723 UTC [TRACE] steampipe: parse complete after 1 decode passes
2023-06-26 15:09:57.723 UTC [TRACE] steampipe: ensureInstallDir /home/patrick/.steampipe
2023-06-26 15:09:57.727 UTC [TRACE] steampipe: nothing to migrate in /home/patrick/.steampipe/internal/update-check.json
2023-06-26 15:09:57.728 UTC [TRACE] steampipe: No memory limit set

2023-06-26 15:09:57.728 UTC [TRACE] steampipe: ociDownloader.Download: downloading us-docker.pkg.dev/steampipe/plugins/turbot/csv:latest
2023-06-26 15:09:57.728 UTC [TRACE] steampipe: ociDownloader.Pull: preparing to pull ref us-docker.pkg.dev/steampipe/plugins/turbot/csv:latest tag latest destDir /home/patrick/.steampipe/plugins/tmp-0e41-4be8-9d66
2023-06-26 15:09:57.728 UTC [TRACE] steampipe: ociDownloader.Pull: pulling...
csv:latest                     [=========>----------------------------------------------------------] Downloading
2023-06-26 15:09:58.426 UTC [ERROR] steampipe: ociDownloader.Pull: failed to pull us-docker.pkg.dev/steampipe/plugins/turbot/csv:latest fail to recognize platform from unknown config applicati
csv:latest                     [====================================================================] Done

Skipped the following plugin:

Plugin:   csv@latest
Reason:   fail to recognize platform from unknown config application/vnd.turbot.steampipe.config.v1+json: expect application/vnd.oci.image.config.v1+json


```

Reply:
@pdecat can you please remove this comment if it's not working/required?